### ISSUE TITLE
Interest in supporting RayStation?

### ISSUE THREAD

**Issue by rabayliss** [cutright/DVH-Analytics#99]
Hi all,
Love the tool. I am testing import of tomotherapy plans produced in RayStation. Unfortunately, during the IMPORT DICOM process, the program seems to be stuck on Parsing File Set 1 of 1. I am running from an executable (hospital workstation), so I don't have much to debug. I am happy to work with you using some anonymized data though.

Thanks!

**Comment by cutright** [comment]
Hi, I'd be happy to add tomo support.  Which version of DVHA are you running, and which OS?

Can you open `~/Apps/dvh_analytics/logs/dvha.log`?  Does it have any contents in it?  Alternatively, if you're on Windows, the command line window stays open... that should print some errors that might be helpful.

If possible, sending me an anonymized DICOM-RT Dose, Plan, and Structure  file set would expedite things. 

Do you only have issues with tomo? Or all of your RS plans?

**Comment by rabayliss** [comment]
Seems like it is all RayStation plans. We are using RS 8.0.b SP2
I am using DVHAnalytics v0.8.7 executing on a windows platform from the compiled executable.

[23540] PyInstaller Bootloader 3.x
[23540] LOADER: executable is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip\DVH Analytics.exe
[23540] LOADER: homepath is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip
[23540] LOADER: _MEIPASS2 is NULL
[23540] LOADER: archivename is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip\DVH Analytics.exe
[23540] LOADER: Extracting binaries
[23540] LOADER: Executing self as child
[23540] LOADER: set _MEIPASS2 to C:\Users\rab034\AppData\Local\Temp\_MEI235402
[23540] LOADER: Setting up to run child
[23540] LOADER: Creating child process
[23540] LOADER: Waiting for child process to finish...
[27364] PyInstaller Bootloader 3.x
[27364] LOADER: executable is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip\DVH Analytics.exe
[27364] LOADER: homepath is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip
[27364] LOADER: _MEIPASS2 is C:\Users\rab034\AppData\Local\Temp\_MEI235402
[27364] LOADER: archivename is C:\Users\rab034\AppData\Local\Temp\Temp1_DVHA-v0.8.7-Windows.zip\DVH Analytics.exe
[27364] LOADER: SetDllDirectory(C:\Users\rab034\AppData\Local\Temp\_MEI235402)
[27364] LOADER: Already in the child - running user's code.
[27364] LOADER: manifestpath: C:\Users\rab034\AppData\Local\Temp\_MEI235402\DVH Analytics.exe.manifest
[27364] LOADER: Error activating the context: ActivateActCtx:
An attempt to set the process default activation context failed because the process default activation context was already set.

[27364] LOADER: ucrtbase.dll is exists: C:\Users\rab034\AppData\Local\Temp\_MEI235402\ucrtbase.dll
[27364] LOADER: Python library: C:\Users\rab034\AppData\Local\Temp\_MEI235402\python37.dll
[27364] LOADER: Loaded functions from Python library.
[27364] LOADER: Manipulating environment (sys.path, sys.prefix)
[27364] LOADER: sys.prefix is C:\Users\rab034\AppData\Local\Temp\_MEI235402
[27364] LOADER: Pre-init sys.path is C:\Users\rab034\AppData\Local\Temp\_MEI235402\base_library.zip;C:\Users\rab034\AppData\Local\Temp\_MEI235402
[27364] LOADER: Setting runtime options
[27364] LOADER: Bootloader option: pyi-windows-manifest-filename DVH Analytics.exe.manifest
[27364] LOADER: Initializing python
[27364] LOADER: Overriding Python's sys.path
[27364] LOADER: Post-init sys.path is C:\Users\rab034\AppData\Local\Temp\_MEI235402\base_library.zip;C:\Users\rab034\AppData\Local\Temp\_MEI235402
[27364] LOADER: Setting sys.argv
[27364] LOADER: setting sys._MEIPASS
[27364] LOADER: importing modules from CArchive
[27364] LOADER: extracted struct
[27364] LOADER: callfunction returned...
[27364] LOADER: extracted pyimod01_os_path
[27364] LOADER: callfunction returned...
[27364] LOADER: extracted pyimod02_archive
[27364] LOADER: callfunction returned...
[27364] LOADER: extracted pyimod03_importers
[27364] LOADER: callfunction returned...
[27364] LOADER: Installing PYZ archive with Python modules.
[27364] LOADER: PYZ archive: PYZ-00.pyz
[27364] LOADER: Running pyiboot01_bootstrap.py
[27364] LOADER: Running pyi_rth__tkinter.py
[27364] LOADER: Running pyi_rth_pkgres.py
[27364] LOADER: Running pyi_rth_mplconfig.py
[27364] LOADER: Running pyi_rth_mpldata.py
[27364] LOADER: Running pyi_rth_multiprocessing.py
[27364] LOADER: Running dvha_app.py

**Comment by rabayliss** [comment]
![image](https://user-images.githubusercontent.com/37907358/97466650-59ffb200-1911-11eb-8530-839a39489024.png)

The import never makes it past the parsing...

**Comment by cutright** [comment]
And nothing in log either? 

If there's nothing in the log I pointed to above, I'd need an anonymized DICOM-RT Plan, Dose, Struct file set to troubleshoot further on my end.

**Comment by rabayliss** [comment]
2020-10-28 10:12:03,703 - dvha - ERROR - Unhandled exception: Traceback (most recent call last):
  File "DVH-Analytics\venv\Lib\site-packages\wx\core.py", line 3285, in <lambda>
wx._core.wxAssertionError: C++ assertion "m_hdc" failed at ..\..\src\msw\textmeasure.cpp(64) in wxTextMeasure::BeginMeasuring(): Must not be used with non-native wxDCs

2020-10-28 10:12:06,996 - dvha - ERROR - Unhandled exception: Traceback (most recent call last):
  File "DVH-Analytics\dvha\main.py", line 1177, in Run
  File "threading.py", line 865, in run
  File "DVH-Analytics\dvha\models\dicom_tree_builder.py", line 686, in do_parse
  File "DVH-Analytics\dvha\models\dicom_tree_builder.py", line 697, in parser
  File "DVH-Analytics\dvha\db\dicom_parser.py", line 84, in __init__
  File "DVH-Analytics\venv\Lib\site-packages\dicompylercore\dicomparser.py", line 802, in GetPlan
  File "DVH-Analytics\venv\Lib\site-packages\pydicom\dataset.py", line 783, in __getattr__
AttributeError: 'FileDataset' object has no attribute 'RTPlanDate'

**Comment by cutright** [comment]
Oh, this is interesting.  This looks like a dicompyler-core issue.

@bastula DVHA executables aren't up to date with master, but I think dicompyler-core master would have this issue too.

https://github.com/dicompyler/dicompyler-core/blob/9595579c2e89d039e1d01f7afd06f5729781496b/dicompylercore/dicomparser.py#L838-L845
```
    def GetPlan(self):
        """Returns the plan information."""


        self.plan = {}


        self.plan['label'] = self.ds.RTPlanLabel
        self.plan['date'] = self.ds.RTPlanDate
        self.plan['time'] = self.ds.RTPlanTime
```

**Comment by cutright** [comment]
Needs testing, but looks like this is fixed here https://github.com/dicompyler/dicompyler-core/commit/42a8cafb80100d3c8d39e090a89e14ba943a9b07

A new dicompyler-core version is planned to be released shortly. Will start testing with DVHA next week.  I'll post a v0.8.8dev1 executable with the latest `dicompyler-core` master early next week.

For users running DVHA from source, you could update `dicompyler-core` to the latest master manually to test it out.

**Comment by cutright** [comment]
@rabayliss I've add a new DVHA pre-release with the latest dicompyler-core master [here](https://github.com/cutright/DVH-Analytics/releases/tag/v0.8.8-dev.1).  Do you mind downloading and trying to reproduce your error?

**Comment by rabayliss** [comment]
Will do. Do you have an address I can send anonymized files?

________________________________

**Comment by cutright** [comment]
@rabayliss I sent you an invite to a private repository with my email address in the README file, if you haven't seen an email notification, might be in your junk folder.  Alternatively, you should be able to find me in the AAPM directory.

**Comment by cutright** [comment]
@rabayliss any progress? It's possible this is resolved with the [v0.8.8 release](https://github.com/cutright/DVH-Analytics/releases/tag/v0.8.8).

**Comment by cutright** [comment]
Based on the information so far, I believe this issue is resolved.  I'm happy to reopen and troubleshoot if new information is provided.
